lib/gosu.rb

```diff
@@ -17,10 +17,374 @@
     path_encoding = ENV["PATH"].encoding
     ENV["PATH"] = "#{binary_path.encode(path_encoding)};#{ENV["PATH"]}"
   end
+
+  # Add the correct lib directory for the current version of Ruby (major.minor).
+  $LOAD_PATH.prepend File.join(binary_path, RUBY_VERSION[/^\d+.\d+/])
 end
 
-require "gosu.#{RbConfig::CONFIG["DLEXT"]}"
+module Gosu
+  extend FFI::Library
+
+  # Load gosu.{so,dylib} which was built by extconf.rb.
+  # It might be useful to also allow loading Gosu installed as a system library, e.g.:
+  # "/usr/local/lib/libgosu-ffi.dylib", # Homebrew on macOS (Intel) or manual installation
+  # "/opt/homebrew/lib/libgosu-ffi.dylib", # Homebrew on macOS (Apple Silicon)
+  # ...or built locally, for testing without having to re-install the gem:
+  # File.expand_path("../../gosu/build/ffi/libgosu-ffi.so", __dir__),
+  # File.expand_path("../../gosu/lib64/gosu-ffi.dll", __dir__),
+  library = "gosu.#{RbConfig::MAKEFILE_CONFIG['DLEXT']}"
+  candidates = ["#{__FILE__}/..", "#{__FILE__}/../../ext/gosu"]
+  candidates.prepend(Gem.loaded_specs['gosu'].extension_dir) if Gem.loaded_specs['gosu']
+  ffi_lib(candidates.map { |dir| File.expand_path(library, dir) })
```

Comment:
Inspired by https://github.com/dbalatero/levenshtein-ffi/pull/11

But I don't want to adjust the ffi_lib paths in every single RB file here. I guess there should only be one central ffi_lib statement.